### PR TITLE
GM: longitudinal support for camera-integrated cars

### PR DESCRIPTION
**Pull request by JasonJShuler**
Add safetyparam and code to enable OP longitudinal on GM vehicles with factory VOACC.

Important: Contains code from PR #962 - There are actually just a handful of changes. Once the PR is merged this can be refreshed from master and it should simmer down to the actual changes.

On cars with camera-based ACC, many users have found OP's VOACC to be superior; additionally without OP long, e2e is not possible.

This provides a parameter that will cause the Panda to block ACC messages from the camera and adjust various conditions (disable controls if ACC messages are present on PT bus, use buttons for enable / disable / speed adjust; allow tx of ACC messages and brake command on PT bus) that enable OP to source the ACC.

The parameter is implemented as a 3rd hardware type for simplicity, but (for now) it only apples to cars with stock VOACC.

### CODE REVIEW

File: board/safety/safety_gm.h [hunk 1]
```diff
@@ -15,13 +15,19 @@ const int GM_MAX_RATE_UP = 7;
...
-                             {0x104c006c, 3, 3}, {0x10400060, 3, 5}};  // gmlan
+
+const CanMsg GM_ASCM_TX_MSGS[] = {{384, 0, 4}, {1033, 0, 7}, {1034, 0, 7}, {715, 0, 8}, {880, 0, 6},  // pt bus
+                                  {161, 1, 7}, {774, 1, 8}, {776, 1, 7}, {784, 1, 2},   // obs bus
+                                  {789, 2, 5},  // ch bus
+                                  {0x104c006c, 3, 3}, {0x10400060, 3, 5}};  // gmlan
+
+const CanMsg GM_CAM_TX_MSGS[] = {{384, 0, 4}};  // pt bus
+
+const CanMsg GM_CAM_VOACC_TX_MSGS[] = {{384, 0, 4}, {715, 0, 8}, {789, 0, 5}, {880, 0, 6}}; // pt bus
```
Comment (sshane) [inline board/safety/safety_gm.h:30]: I'd change VOACC to LONG or LONGITUDINAL

Comment (sshane) [inline board/safety/safety_gm.h:30]:
```suggestion
const CanMsg GM_CAM_LONG_TX_MSGS[] = {{384, 0, 4}, {715, 0, 8}, {789, 0, 5}, {880, 0, 6}}; // pt bus
```

File: board/safety/safety_gm.h [hunk 2]
```diff
@@ -16,12 +16,30 @@ const int GM_MAX_RATE_DOWN = 17;
...
+};
+
+const GmLongLimits GM_CAM_LONG_LIMITS = {
+  .max_gas = 3400,
+  .max_regen = 1514,
+  .inactive_regen = 1554,
+  .max_brake = 400,
+};
+
+const GmLongLimits *GM_LONG_LIMITS;
```
Comment (adeebshihadeh) [inline board/safety/safety_gm.h:40]: make the name lower case

File: tests/safety/test_gm.py [hunk 3]
```diff
@@ -200,5 +208,30 @@ def test_gas_safety_check(self):
...
+    self.packer = CANPackerPanda("gm_global_a_powertrain_generated")
+    self.packer_chassis = CANPackerPanda("gm_global_a_chassis")
+    self.safety = libpandasafety_py.libpandasafety
+    self.safety.set_safety_hooks(Panda.SAFETY_GM, Panda.FLAG_GM_HW_CAM | Panda.FLAG_GM_HW_CAM_LONG)
+    self.safety.init_tests()
+
+  # brake message sends on bus 0 instead of 2 for GM Camera cars
+  def _send_brake_msg(self, brake):
+    values = {"FrictionBrakeCmd": -brake}
+    return self.packer_chassis.make_can_msg_panda("EBCMFrictionBrakeCmd", 0, values)
```
Comment (adeebshihadeh) [inline tests/safety/test_gm.py:233]: add a variable for bus instead of redefining the function

**Pull request merged**